app/guard/guard.go

```diff
@@ -0,0 +1,256 @@
+// Guard is a service that provides additional password protection by creating a series of guard servers that allow a
+// Client to further secure their passwords from direct database compromises. The service is hash based and the passwords
+// never leave the main database, making the guard servers very lightweight. The guard server's are used in both setting and
+// authenticating passwords.
+
+package main
+
+import (
+	"os"
+
+	"errors"
+
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+
+	"github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/cothority/services/guard"
+	"github.com/dedis/crypto/abstract"
+
+	"io/ioutil"
+
+	s "github.com/SSSaaS/sssa-golang"
+	"gopkg.in/codegangsta/cli.v1"
+
+	"bytes"
+
+	"github.com/dedis/cothority/network"
+)
+
+// User is a representation of the Users data in the code, and is used to store all relevant information
+type User struct {
+	// Name or UserID
+	Name []byte
+	// Salt used for the password-hash
+	Salt []byte
+	// Xored keys with hash
+	Keys [][]byte
+	// Data AEAD-encrypted with key
+	Data []byte
+}
+
+// Database is a structure that stores Cothority(the list of guard servers), and a list of all users within the database
+type Database struct {
+	Cothority *sda.Roster
+	Users     []User
+}
+
+var db *Database
+
+var mastersalt []byte
```

Comment:
`mastersalt` is only used in `set`, so initialize it there. It doesn't need to be global.